subscriptions.go

```diff
@@ -265,7 +265,7 @@ type SubscriptionNotes struct {
 	TermsAndConditions    string        `xml:"terms_and_conditions,omitempty"`
 	CustomerNotes         string        `xml:"customer_notes,omitempty"`
 	VATReverseChargeNotes string        `xml:"vat_reverse_charge_notes,omitempty"`
-	GatewayCode           string        `xml:"gateway_code,omitempty"`
+	GatewayCode           *string       `xml:"gateway_code,omitempty"`
```

Comment:
I dug through the repo and don't believe we use any non-struct pointer types. I'd like to keep that consistent, as well as make it simple for library users to check if the `GatewayCode` is empty (vs also having to check that it's non-nil first).

Let's keep this as `string` and instead implement a basic `MarshalXML` handler that omits encoding the `gateway_code` tag if `GatewayCode` is an empty string so this is just an implementation detail of the library.

Reply:
ok, I've added a NullString type, please take a look. thanks